services/web/server/src/simcore_service_webserver/snapshots_api_handlers.py

```diff
@@ -0,0 +1,255 @@
+import logging
+from datetime import datetime
+from functools import wraps
+from typing import Any, Callable, List, Optional
+from uuid import UUID
+
+import orjson
+from aiohttp import web
+from pydantic.decorator import validate_arguments
+from pydantic.error_wrappers import ValidationError
+from pydantic.main import BaseModel
+from yarl import URL
+
+from ._meta import api_version_prefix as vtag
+from .constants import RQ_PRODUCT_KEY, RQT_USERID_KEY
+from .login.decorators import login_required
+from .projects import projects_api
+from .projects.projects_exceptions import ProjectNotFoundError
+from .security_decorators import permission_required
+from .snapshots_core import ProjectDict, take_snapshot
+from .snapshots_db import ProjectsRepository, SnapshotsRepository
+from .snapshots_models import Snapshot, SnapshotItem
+
+logger = logging.getLogger(__name__)
+
+
+def _default(obj):
+    if isinstance(obj, BaseModel):
+        return obj.dict()
+    raise TypeError
+
+
+def json_dumps(v) -> str:
+    # orjson.dumps returns bytes, to match standard json.dumps we need to decode
+    return orjson.dumps(v, default=_default).decode()
+
+
+def enveloped_response(data: Any, **extra) -> web.Response:
+    enveloped: str = json_dumps({"data": data, **extra})
+    return web.Response(text=enveloped, content_type="application/json")
+
+
+def handle_request_errors(handler: Callable):
```

Comment:
Can you give the signature of the `handler` e.g. `Callable[[str, Any], Optional[str]]`?

Reply:
i changed to
```python
def handle_request_errors(handler: Callable[[web.Request], Awaitable[web.Response]]):
 ...
```